### PR TITLE
Added type hints for TF: rag model

### PR DESCRIPTION
**Pull request by debjit-bw**
Based on Issue https://github.com/huggingface/transformers/issues/16059

Type hints for the [TFRagModel](https://huggingface.co/docs/transformers/model_doc/rag) have been added.

@Rocketknight1 Could you please check the changes and merge if it's fine?

Thanks a lot.

### CODE REVIEW

**Comment by debjit-bw** [comment]
@Rocketknight1 can you take a look at this? I just added type hints and fixed a syntax error in one previous type. Still it shows it failed these 2 tests. I have no idea what these tests do, can you shed some light on this issue?

File: src/transformers/models/rag/modeling_tf_rag.py [hunk 1]
```diff
@@ -538,22 +538,22 @@ def set_retriever(self, retriever: RagRetriever):
...
-        context_input_ids=None,
-        context_attention_mask=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        output_retrieved=None,
-        n_docs=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/rag/modeling_tf_rag.py:541]:
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

File: src/transformers/models/rag/modeling_tf_rag.py [hunk 2]
```diff
@@ -828,25 +828,25 @@ def marginalize(self, seq_logits, doc_scores, n_docs=None):
...
-        output_attentions=None,
-        output_hidden_states=None,
-        output_retrieved=None,
-        n_docs=None,
-        do_marginalize=None,
-        labels=None,
-        reduce_loss=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/rag/modeling_tf_rag.py:831]:
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

File: src/transformers/models/rag/modeling_tf_rag.py [hunk 3]
```diff
@@ -1425,27 +1425,27 @@ def question_encoder(self):
...
-        output_attentions=None,
-        output_hidden_states=None,
-        output_retrieved=None,
-        n_docs=None,
-        exclude_bos_score=None,
-        labels=None,
-        reduce_loss=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[tf.Tensor] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/rag/modeling_tf_rag.py:1428]:
```suggestion
        input_ids: Optional[TFModelInputType] = None,
```

**Review by Rocketknight1 — APPROVED** [review]
Overall this looks good! Don't worry about the test - that seems totally unrelated. You could probably fix it by pulling upstream commits to your fork's `main` branch and then rebasing your PR branch onto the main branch followed by a force-push, but don't worry if you're not sure how to do that - I'm happy to merge even with those tests failing.

My one issue is that the first input type for TF models should be `TFModelInputType`. You'll need to import that type from `modeling_tf_utils.py`.

**Comment by debjit-bw** [comment]
Thanks a lot @Rocketknight1, I have made the suggested changes!

**Pull request merged**